bftengine/src/bftengine/ReplicaImp.hpp

```diff
@@ -118,6 +118,23 @@ class ReplicaImp : public InternalReplicaApi, public ReplicaForStateTransfer {
   std::map<uint64_t, std::pair<Time, ClientRequestMsg*>>
       requestsOfNonPrimary;  // used to retransmit client requests by a non primary replica
   size_t NonPrimaryCombinedReqSize = 1000;
+  //
+  const std::thread::id MAIN_THREAD_ID;
+  uint16_t activeExecutions_ = 0;                     // represents the number of running executions at this moment.
```

Comment:
Your'e right its kind of boolean but I'll keep it like this for calculations like: `isSeqNumToStopAt(lastExecutedSeqNum + activeExecutions_ + 1)` instead of casting a bool.

Added explanation to the documentation.